connectors/src/connectors/github/temporal/cast_known_errors.ts

```diff
@@ -17,17 +31,12 @@ export class GithubCastKnownErrorsInterceptor
     try {
       return await next(input);
     } catch (err: unknown) {
-      if (err instanceof RequestError) {
-        if (err.status === 403 && err.name === "HttpError") {
-          throw new ProviderWorkflowError(
-            "github",
-            "403 - Rate Limit Error",
-            "rate_limit_error",
-            err
-          );
-        }
+      checkRateLimitError(err);
+      if (err instanceof WithRetriesError) {
+        err.errors.forEach((error) => {
+          checkRateLimitError(error);
+        });
```

Comment:
Can they be nested more than one level deep? If so, maybe move the WithRetriesError to the helper method to make it recursive.

Reply:
No , they should not be nested